app/controllers/user/base_controller.rb

```diff
@@ -8,4 +9,10 @@ def verify_google_user
       render 'error/unauthorized'
     end
   end
+
+  def no_api_response
+    @current_module = TuringModule.find(params[:turing_module_id])
+    flash[:error] = "The page you are trying to access is currently unavailable. This may be due to an API call failing. Please try again later."
```

Comment:
"This may be due to an API call failing." This may imply that the user made a mistake. Maybe we can make this message a bit more general.

Reply:
In general this might be bad practice on most apps, but given the target customers here are instructors, you could almost display the request that was made and the raw error you got (almost kind of like debug mode in Rails). You wouldn't want to output secrets like an API key, but you could be very transparent to the user about the URL and the arguments in the request, which might point the instructors to the issue. Not sure if that's something easy to implement here, but just an idea.

Reply:
I think in this scenario It would be a bit tough to display the request parameters and error messages to the API. This is because when the API returns an error or no response, it leads to a failure (ActionView::Template:Error or NoMethodError) in our application down the line. That is the internal Present methods that are relying on the API data error out. When they error out, there is no data being held that provides details on the API call that led to that failure. 

I think there are probably ways to write code that lead to good logging of which API calls failed, but in this scenario I would have to do research to learn how to implement them.

Reply:
@AlecKap I went ahead and changed the message to be more generic. Let me know if you like it!

Reply:
Yeah looks good to me! Seems more of an all encompassing message.